### PR TITLE
Add degreeOfParallelismPercent to ADLA Job Contract

### PR DESCRIPTION
**Pull request by hosunmsft**
This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process.

### PR information
- [x] The title of the PR is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).
- [x] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [x] If applicable, the PR references the bug/issue that it fixes.
- [x] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [x] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [ ] My spec meets the review criteria:
  - [x] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [x] The spec follows the guidelines described in the [Swagger checklist](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md) document.
  - [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

File: specification/datalake-analytics/data-plane/Microsoft.DataLakeAnalytics/stable/2016-11-01/job.json [hunk 1]
```diff
@@ -1483,7 +1488,12 @@
           "type": "integer",
           "default": 1,
           "format": "int32",
-          "description": "the degree of parallelism used for this job. This must be greater than 0, if set to less than 0 it will default to 1."
+          "description": "the degree of parallelism used for this job. At most one of degreeOfParallelism and degreeOfParallelismPercent should be specified. If none, a default value of 1 will be used."
+        },
+        "degreeOfParallelismPercent": {
+          "type": "number",
+          "format": "double",
+          "description": "the degree of parallelism in percentage used for this job. At most one of degreeOfParallelism and degreeOfParallelismPercent should be specified. If none, a default value of 1 will be used for degreeOfParallelism."
```
Comment (anuchandy) [inline specification/datalake-analytics/data-plane/Microsoft.DataLakeAnalytics/stable/2016-11-01/job.json]: I don't see JobInformationBasic or any type deriving from it being used as PUT|PATCH|POST request payload. So far they seems used to represent data returned from the service, please correct me if I'm wrong. If so should we mark these properties as readonly in JobInformationBasic model?

Comment (hosunmsft) [reply specification/datalake-analytics/data-plane/Microsoft.DataLakeAnalytics/stable/2016-11-01/job.json]: You are right Anu, I just updated the change, please take another look. Thanks.

**Comment by anuchandy** [comment]
@AutorestCI regenerate azure-sdk-for-node

**Review by anuchandy — APPROVED** [review]

**Comment by anuchandy** [comment]
Ran linter locally and it's green. CI failures are unrelated to these changes hence not holding the PR.  Team is investigating the CI failures.

**Pull request merged**